File: AAMFeedback/ViewController.m
//
//  ViewController.m
//  AAMFeedbackViewController
//
//  Created by 深津 貴之 on 11/11/30.
//  Copyright (c) 2011年 Art & Mobile. All rights reserved.
//

#import "ViewController.h"
#import "AAMFeedbackViewController.h"

@implementation ViewController


- (IBAction)handleOpenAsModal:(id) sender {
    AAMFeedbackViewController *vc = [[AAMFeedbackViewController alloc] init];
    vc.toRecipients = @[@"YOUR_CONTACT@email.com"];
    vc.ccRecipients = nil;
    vc.bccRecipients = nil;
    vc.view.backgroundColor = [UIColor whiteColor];
    vc.beforeShowAction = ^(MFMailComposeViewController *controller) {
        [controller addAttachmentData:[@"text" dataUsingEncoding:NSUTF8StringEncoding] mimeType:@"text/plain" fileName:@"example.text"];
    };
    vc.descriptionPlaceHolder = @"Please write for details in modal.";
    UINavigationController *feedbackNavigation = [[UINavigationController alloc] initWithRootViewController:vc];
    [self presentViewController:feedbackNavigation animated:YES completion:nil];
}
@end

File: Feedback/ViewController.swift
//
//  ViewController.swift
//  Feedback
//
//  Created by akuraru on 2020/04/12.
//  Copyright © 2020 Art & Mobile. All rights reserved.
//

import UIKit

class ViewController: UIViewController {

    override func viewDidLoad() {
        super.viewDidLoad()
    }

    @IBAction func openModal(_ sender: Any) {
        let viewController = createViewController()
        let feedbackNavigation = UINavigationController(rootViewController: viewController)
        present(feedbackNavigation, animated: true)
        
    }
    @IBAction func pushAsViewController(_ sender: Any) {
        let viewController = createViewController()
        navigationController?.pushViewController(viewController, animated: true)
    }

    func createViewController() -> UIViewController {
        let viewController = FeedbackViewController()
        viewController.context = Context(
            toRecipients: ["YOUR_CONTACT@email.com"],
            descriptionPlaceHolder: "Please write for details in modal."
        )
        viewController.beforeShowAction = { (controller) in
            controller.addAttachmentData("text".data(using: .utf8)!, mimeType: "text/plain", fileName: "example.text")
        }
        viewController.view.backgroundColor = .white
        return viewController
    }
}

